proxy/http/HttpSM.cc

```diff
@@ -6626,6 +6630,9 @@ HttpSM::setup_blind_tunnel(bool send_response_hdr)
   milestones[TS_MILESTONE_SERVER_BEGIN_WRITE] = Thread::get_hrtime();
   if (send_response_hdr) {
     client_response_hdr_bytes = write_response_header_into_buffer(&t_state.hdr_info.client_response, to_ua_buf);
+    if (initial && initial->read_avail()) {
```

Comment:
I have added the consume data as suggested. I will need more context around the CONNECT method code change. 
When developing the fix I had a very targeted bug which was easy to reproduce, hence I only fixed the initial buffer scenario. My automated test will not cover extension to the CONNECT method, but I will be able to determine if the suggested change breaks things.